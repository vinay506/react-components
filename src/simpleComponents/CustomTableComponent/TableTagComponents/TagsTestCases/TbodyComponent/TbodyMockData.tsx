import React from 'react';
import { TextWithTooltip } from '../../../../../compoundComponents';
import { InfoIcon } from '../../../../index';


const rows = [{
    column1: <TextWithTooltip text='Chapter Test' tooltip={{ id: 'chapterTest', backgroundColor: '#d8c7c7', textColor: '#000', multiline: true, place: 'bottom' }} tooltipText="Chapter Test" ><InfoIcon styles={{ width: '20px', height: '15px' }}></InfoIcon></TextWithTooltip>,
    column2: 'Selected Chapters (30)',
    column3: 'All Chapters (91)',
    column4: 'All Chapters (91)'
}, {
    column1: <TextWithTooltip text='Part Syllabus Test' tooltip={{ id: 'partSyllabusTest', backgroundColor: '#d8c7c7', textColor: '#000', multiline: true, place: 'bottom' }} tooltipText="Part Syllabus Test" ><InfoIcon styles={{ width: '20px', height: '15px' }}></InfoIcon></TextWithTooltip>,
    column2: '3',
    column3: '20',
    column4: '20'
}, {
    column1: <TextWithTooltip text='Full Syllabus Test' tooltip={{ id: 'fullSyllabusTest', backgroundColor: '#d8c7c7', textColor: '#000', multiline: true, place: 'bottom' }} tooltipText="Full Syllabus Test"><InfoIcon styles={{ width: '20px', height: '15px' }}></InfoIcon></TextWithTooltip>,
    column2: '3',
    column3: '6',
    column4: '6'
}, {
    column1: <TextWithTooltip text='Past Year Papers' tooltip={{ id: 'pastYearPapers', backgroundColor: '#d8c7c7', textColor: '#000', multiline: true, place: 'bottom' }} tooltipText="Past Year Papers"><InfoIcon styles={{ width: '20px', height: '15px' }}></InfoIcon></TextWithTooltip>,
    column2: '20',
    column3: '20',
    column4: '20'
}]

const bodyColumns = [{ key: "column1", styles: { fontWeight: 'bold', height: '45px' } }, { key: "column2", styles: { height: '45px' } },
{ key: "column3", styles: { height: '45px' } }, { key: "column4", styles: { height: '45px' } }];

export { rows, bodyColumns }