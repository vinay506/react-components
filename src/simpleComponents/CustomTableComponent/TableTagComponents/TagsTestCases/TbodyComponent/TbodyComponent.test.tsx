import React from 'react';
import { shallow } from 'enzyme';
import { CustomTbody,CustomTableRow,CustomTdCell } from '../../index';
import renderer from 'react-test-renderer'
import {rows,bodyColumns} from './TbodyMockData';

const getCustomTableRows = (rows: Array<any>) => {
    return rows.map((row,index) => {
      return getCustomTableRow(row,index)
    })
  }

  const getCustomTableRow = (row: any,index:number) => {
    return (
      <CustomTableRow key={'row-'+index}>
        {getCustomTableCells(row)}
      </CustomTableRow>
    )
  }

  const getCustomTableCells = (row: any) => {
    return (bodyColumns || []).map((column: any,index:number) => {
      return <CustomTdCell style={column.styles} key={'bdcol-'+index}>{row[column.key]}</CustomTdCell>
    })
  }

describe('<Custom TBody Tag >', function () {
    let props: any;
    let tree: any;
    let wrapper:any;
    beforeEach(() => {
        props = {};
        wrapper = shallow(<CustomTbody {...props}>{getCustomTableRows(rows)}</CustomTbody>);
    });

    // Snapshot test
    it('should render', () => {
        tree = renderer.create(<CustomTbody {...props}></CustomTbody>).toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('checks with the className of Table body',() => {
        const props = wrapper.find('tbody').props();
        expect(props.className).toEqual('mdc-data-table__content')
    })
    it('should render Custom Table rows cell', () => {
        const customTableRow = wrapper.find(CustomTableRow);
        expect(customTableRow.exists()).toBe(true);    
    })

    it('checks Number of records it is rendering', () => {
        const customTableRow = wrapper.find(CustomTableRow);
        expect(customTableRow).toHaveLength(4);    
    })

   
    
})