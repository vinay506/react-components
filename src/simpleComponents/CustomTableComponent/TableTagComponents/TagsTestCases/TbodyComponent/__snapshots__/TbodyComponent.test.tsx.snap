// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Custom TBody Tag > should render 1`] = `
<tbody
  className="mdc-data-table__content"
/>
`;
